app/src/ui/clone-repository/clone-repository.tsx

```diff
@@ -330,23 +342,40 @@ export class CloneRepository extends React.Component<
       accounts.push(this.props.enterpriseAccount)
     }
 
-    const account = await findAccountForRemoteURL(url, accounts)
+    const account = await findAccountForRemoteURL(this.state.url, accounts)
+
     if (identifier && account) {
       const api = API.fromAccount(account)
       const repo = await api.fetchRepository(identifier.owner, identifier.name)
       if (repo) {
```

Comment:
Changing this up to return immediately does become a bit easier to understand from the control flow (which isn't great already):

```ts
if (repo) {
  return {
    url: repo.clone_url,
    friendlyName: repo.name,
  }
}
```

Reply:
Done :)